### DISCUSSION TITLE
Resume search / repeat last search

### DISCUSSION THREAD

**Discussion by Pacane** [minad/consult#866] [Q&A] [answered]
Hi I'm trying to find out if there's a way to preserve the last search terms in between consult usages.

For example I do a lot of ripgrep searches using consult and navigate to individual results. Is there a way to search for the same terms again without having to type the whole thing again?

**Comment by oantolin** [comment, marked as answer]
The search terms get added to the history, so `M-p` after restarting the search command.

**Comment by minad** [comment]
There also the commands `vertico-repeat` and `vertico-suspend`. But I usually recommend to use `embark-collect` or `embark-export` to avoid unnecessarily restarting the search.

**Reply by Pacane** [reply]
I couldn't get vertico-repeat/suspend to work with consult. Is it supposed to work and is there docs for it?

**Reply by hjudt** [reply]
I am happily using vertico-suspend. Just bind it to a key, no further configuration required. But embark-export/-collect are usually the more convenient approach.
